### PR TITLE
[BE] APDS-6 - make intercom a standalone package

### PR DESCRIPTION
**Pull request by poxip**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2016 Arabella
```
Comment (remik) [inline LICENSE]: It is 2017 already.

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+**aa-intercom** requires a few fields in the user model to be set. To make it work, you need to apply the ``aa_intercom.mixins.IntercomUserMixin`` to your custom user model (if you do not have your own custom user model set, check the `documentation <https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#specifying-a-custom-user-model>`_).
+
+Configuration
+-------------
+The last thing is to specify Intercom credentials
+
+::
+
+  INTERCOM_APP_ID = "app id"
+  INTERCOM_ID_PREFIX = ""
```
Comment (remik) [inline README.rst]: I think prefix is not credentials but some of our settings.

File: aa_intercom/management/commands/update_intercom_users.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+
+
+class Command(BaseCommand):
+    help = "Update all users in intercom.io"
+
+    def handle(self, **options):
+        # return  # not sure if it is not needed. Commenting out for now to check.
```
Comment (remik) [inline aa_intercom/management/commands/update_intercom_users.py]: ??

File: aa_intercom/migrations/0002_auto_20170224_1455.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (remik) [inline aa_intercom/migrations/0002_auto_20170224_1455.py]: Can we have only 1 migration before release?

File: aa_intercom/models.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+from django.db import models
+from django.db.models.signals import post_save
+from django.utils.translation import ugettext_lazy as _
+
+from aa_intercom.signals import intercom_event_push_to_intercom_post_save
+
+
+class IntercomEvent(models.Model):
+    """
+        Fill types per project.
```
Comment (remik) [inline aa_intercom/models.py]: This is not abstract so how we can fill it? Probably types should be in settings.

File: requirements/requirements-base.txt [hunk 6]
```diff
@@ -0,0 +1,2 @@
+python-intercom==3.1.0
```
Comment (remik) [inline requirements/requirements-base.txt]: Do not freeze version in libraries

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+
+requirements = local_open("requirements/requirements-base.txt")
+required_to_install = [dist.strip() for dist in requirements.readlines()]
+
+
+setup(
+    name="aa_intercom",
+    version=version,
+    url="https://github.com/ArabellaTech/aa_intercom",
```
Comment (remik) [inline setup.py]: typo

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+---------------------
+**aa-intercom** requires a few fields in the user model to be set. To make it work, you need to apply the ``aa_intercom.mixins.IntercomUserMixin`` to your custom user model (if you do not have your own custom user model set, check the `documentation <https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#specifying-a-custom-user-model>`_).
+
+Configuration
+-------------
+The last thing is to specify Intercom credentials
+
+::
+
+  INTERCOM_APP_ID = "app id"
```
Comment (remik) [inline README.rst]: It is not used.

File: aa_intercom/admin.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+from django.contrib import admin
+
+from aa_intercom.models import IntercomEvent
+
+
+class IntercomEventAdmin(admin.ModelAdmin):
```
Comment (remik) [inline aa_intercom/admin.py]: This should be read only.

File: README.rst [hunk 10]
```diff
@@ -0,0 +1,56 @@
+===========
+aa-intercom
+===========
+|travis|_ |pypi|_ |coveralls|_ |requiresio|_
+
+Django integration for Intercom_ API
```
Comment (remik) [inline README.rst:6]: Please describe what is it doing and how to use it. We can use it for more the user model, right?

**Review by remik — CHANGES_REQUESTED** [review]
Please use `master` as main branch. We don't use git flow for libraries.

File: aa_intercom/management/commands/update_intercom_users.py (hunk 3 above)
Comment (poxip) [reply aa_intercom/management/commands/update_intercom_users.py]: That's old @jacoor's code. @jacoor should we drop it?

File: README.rst [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+Configuration
+-------------
+The last thing is to specify Intercom credentials
+
+::
+
+  INTERCOM_APP_ID = "app id"
+  INTERCOM_ID_PREFIX = ""
+  INTERCOM_API_ACCESS_TOKEN = "your access key"
```
Comment (jacoor) [inline README.rst]: access token not key

File: aa_intercom/management/commands/update_intercom_users.py (hunk 3 above)
Comment (jacoor) [reply aa_intercom/management/commands/update_intercom_users.py]: @poxip please drop it completely.

File: aa_intercom/migrations/0002_auto_20170224_1455.py (hunk 4 above)
Comment (jacoor) [reply aa_intercom/migrations/0002_auto_20170224_1455.py]: +1. Please remove all migrations and rerun migrate to have only one.

File: aa_intercom/utils.py [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+from django.contrib.auth import get_user_model
+from django.utils.timezone import make_aware, now
+from intercom import errors
+from intercom.client import Client
+
+intercom = Client(personal_access_token=settings.INTERCOM_API_ACCESS_TOKEN)
+
+
+def upload_intercom_user(obj_id):
+    """Creates or updates single user account on intercom
```
Comment (jacoor) [inline aa_intercom/utils.py]: Please move it to new line, alignt with """ or move to one line w/o spaces between. """ and text.

File: test_project/urls.py [hunk 13]
```diff
@@ -0,0 +1,16 @@
+"""test_project URL Configuration
+
+The `urlpatterns` list routes URLs to views. For more information please see:
```
Comment (jacoor) [inline test_project/urls.py]: Do we really need this comment?

File: test_project/wsgi.py [hunk 14]
```diff
@@ -0,0 +1,16 @@
+"""
+WSGI config for test_project project.
```
Comment (jacoor) [inline test_project/wsgi.py]: As above, do we need this comment?

**Review by jacoor — CHANGES_REQUESTED** [review]

File: aa_intercom/management/commands/update_intercom_users.py (hunk 3 above)
Comment (poxip) [reply aa_intercom/management/commands/update_intercom_users.py]: Ok

**Review by remik — APPROVED** [review]

**Review by jacoor — APPROVED** [review]

**Pull request merged**